### PR TITLE
FIX: Ensure that custom {{action}} modifier works with actions hash

### PR DESCRIPTION
**Pull request by dgeb**
A callback that's provided as a string, such as `{{action "doSomething"}}`, may target the method `doSomething` on the context OR the context's `action` hash (if it exists).

### CODE REVIEW

File: app/assets/javascripts/discourse/app/lib/ember-action-modifier.js [hunk 1]
```diff
@@ -16,7 +16,12 @@ export const actionModifier = modifier(
     { on, bubbles, preventDefault, allowedKeys }
   ) => {
     const handler = (event) => {
-      const fn = typeof callback === "string" ? context[callback] : callback;
+      let fn;
+      if (typeof callback === "string") {
+        fn = context[callback] ?? context.actions?.[callback];
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/lib/ember-action-modifier.js:21]: It's possible for both `context[callback]` and `context.actions[callback]` to exist - do we know which way around Ember itself prioritises them? My gut says it's probably `actions[callback]` first, but I'm not sure 🤔

Comment (dgeb) [reply app/assets/javascripts/discourse/app/lib/ember-action-modifier.js:21]: Your gut is right :)

I'll add tests to ensure that the correct precedence is used.

Comment (dgeb) [reply app/assets/javascripts/discourse/app/lib/ember-action-modifier.js]: @davidtaylorhq ok, updated with precedence fix. Good catch!

**Review by davidtaylorhq — APPROVED** [review]

**Comment by davidtaylorhq** [comment]
Looks great, thanks for the quick fix @dgeb

**Pull request merged**